src/App.tsx

```diff
@@ -15,36 +15,114 @@
+  const [activeSort, setActiveSort] = useState<SortType>(SortType.Default);
+
+  const resetGoods = () => {
+    setGoods([...goodsFromServer]);
+    setIsReversed(false);
+    setActiveSort(SortType.Default);
+  };
+
+  const sortAlphabetically = () => {
+    setGoods(() => {
+      const sortedGoods = [...goodsFromServer].sort((a, b) =>
+        a.localeCompare(b),
+      );
+
+      return isReversed ? sortedGoods.reverse() : sortedGoods;
+    });
+    setActiveSort(SortType.Alphabetical);
+  };
+
+  const sortByLength = () => {
+    setGoods(() => {
+      const sortedGoods = [...goodsFromServer].sort(
+        (a, b) => a.length - b.length,
+      );
+
+      return isReversed ? sortedGoods.reverse() : sortedGoods;
+    });
+    setActiveSort(SortType.Length);
+  };
+
+  const reverseGoods = () => {
+    setGoods(prevGoods => [...prevGoods].reverse());
+    setIsReversed(prevReversed => !prevReversed);
+  };
+
+  let alphabeticalButtonClass = 'button is-info is-light';
+
+  if (activeSort === SortType.Alphabetical) {
+    alphabeticalButtonClass = 'button is-info';
+  }
+
+  let lengthButtonClass = 'button is-success is-light';
+
+  if (activeSort === SortType.Length) {
+    lengthButtonClass = 'button is-success';
+  }
+
+  let reverseButtonClass = 'button is-warning is-light';
+
+  if (isReversed) {
+    reverseButtonClass = 'button is-warning';
+  }
+
+  const resetButton =
+    goods.join() !== goodsFromServer.join() ? (
+      <button
+        type="button"
+        className="button is-danger is-light"
+        onClick={resetGoods}
+      >
+        Reset
+      </button>
+    ) : null;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={alphabeticalButtonClass}
+          onClick={sortAlphabetically}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={lengthButtonClass}
+          onClick={sortByLength}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={reverseButtonClass}
+          onClick={reverseGoods}
+        >
           Reverse
         </button>
-
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {resetButton}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {goods.map(good => (
+          <li key={good} data-cy="Good">
```

Comment:
The `data-cy` attribute is used for testing purposes. Ensure that this attribute is necessary for your testing framework and that it is correctly implemented according to your testing strategy.